x/metadata/keeper/msg_server.go

```diff
@@ -48,6 +59,44 @@ func (k msgServer) WriteScope(
 	return types.NewMsgWriteScopeResponse(msg.Scope.ScopeId), nil
 }
 
+func (k msgServer) createExpirationForScope(ctx sdk.Context, msg *types.MsgWriteScopeRequest) (*exptypes.Expiration, error) {
```

Comment:
This not only creates the expiration, but also sets it and charges the deposit. It would be better named `setExpirationForScope` and only returning an `error`. The `Expiration` return value is never used anyway.

Reply:
renamed to `setExpirationForScope` and changed method signature to return only `error`.